### PR TITLE
EMR Script support for the PyPi, Maven, NPM

### PR DESCRIPTION
**Pull request by ravsa**

### CODE REVIEW

File: rudra/deployments/emrs/pypi_emr.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
+"""EMR script implementation for the PYPI service."""
```
Comment (miteshvp) [inline rudra/deployments/emrs/pypi_emr.py]: All these EMR files looks the same. Can this not be clubbed further?

File: rudra/deployments/emrs/emr_script_builder.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+                             .format(missing_fields))
+
+        self.env = input_dict.get('environment')
+        user, repo = get_github_repo_info(input_dict.get('github_repo'))
+        self.training_file_url = get_training_file_url(user, repo)
+        self.aws_access_key = os.getenv("AWS_S3_ACCESS_KEY_ID") \
+            or input_dict.get('aws_access_key')
+        self.aws_secret_key = os.getenv("AWS_S3_SECRET_ACCESS_KEY")\
+            or input_dict.get('aws_secret_key')
+        self.bucket_name = input_dict.get('bucket_name')
```
Comment (miteshvp) [inline rudra/deployments/emrs/emr_script_builder.py]: We can check for not None here. Else fail the script

File: rudra/deployments/emrs/emr_script_builder.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+        missing_fields = check_field_exists(input_dict, required_fields)
+
+        if missing_fields:
+            logger.error("Missing the parameters in input_dict",
+                         extra=missing_fields)
+            raise ValueError("Required fields are missing in the input {}"
+                             .format(missing_fields))
+
+        self.env = input_dict.get('environment')
```
Comment (miteshvp) [inline rudra/deployments/emrs/emr_script_builder.py]: We can check for non-null value here. If None, then fail the script

File: rudra/deployments/emrs/pypi_emr.py (hunk 1 above)
Comment (ravsa) [reply rudra/deployments/emrs/pypi_emr.py]: Yes, for now,  EMR files looks the same but we have to customize these files according to the model requirements.

File: rudra/deployments/emrs/emr_script_builder.py (hunk 2 above)
Comment (ravsa) [reply rudra/deployments/emrs/emr_script_builder.py]: added validation in utils module, Thanks

File: rudra/deployments/emrs/emr_script_builder.py (hunk 3 above)
Comment (ravsa) [reply rudra/deployments/emrs/emr_script_builder.py]: added

**Pull request merged**